# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/GLImplFactory.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/renderer/GLImplFactory.h
--- a/gfx/angle/checkout/src/libANGLE/renderer/GLImplFactory.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/GLImplFactory.h	Sat Sep 07 13:38:36 2019 +0000
@@ -35,12 +35,14 @@
 class FenceNVImpl;
 class SyncImpl;
 class FramebufferImpl;
+class MemoryObjectImpl;
 class PathImpl;
 class ProgramImpl;
 class ProgramPipelineImpl;
 class QueryImpl;
 class RenderbufferImpl;
 class SamplerImpl;
+class SemaphoreImpl;
 class ShaderImpl;
 class TextureImpl;
 class TransformFeedbackImpl;
@@ -88,6 +90,12 @@
     virtual ProgramPipelineImpl *createProgramPipeline(const gl::ProgramPipelineState &data) = 0;
 
     virtual std::vector<PathImpl *> createPaths(GLsizei range) = 0;
+
+    // Memory object creation
+    virtual MemoryObjectImpl *createMemoryObject() = 0;
+
+    // Semaphore creation
+    virtual SemaphoreImpl *createSemaphore() = 0;
 };
 
 }  // namespace rx